### PR TITLE
Migrate cloudformation-guard to GitHub actions

### PR DESCRIPTION
**Pull request by jotompki**
*Issue #, if available:*

*Description of changes:* Migrating off TravisCI. This job simply builds the entire rust workspace and runs unit tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  pull_request:
+    branches: [ master ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (PatMyron) [inline .github/workflows/rust.yml:15]: Does this work on `macos-latest` and `windows-latest`? Can we add those if it does?

Comment (jotompki) [reply .github/workflows/rust.yml:15]: good looks

**Review by omkhegde — APPROVED** [review]

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: cargo build --verbose --release
+    - name: Run tests
+      run: cargo test --verbose
```
Comment (PatMyron) [inline .github/workflows/rust.yml:24]: Do we need to run these in each directory still?
https://github.com/aws-cloudformation/cloudformation-guard/pull/64/files

Comment (jotompki) [reply .github/workflows/rust.yml:24]: i set this up as a rust workspace so it builds everything in the directory (including sub directories). check the logs for one of the operating systems below to see that the project is building all the packages

**Review by PatMyron — APPROVED** [review]

**Comment by PatMyron** [comment]
Could get rid of the Travis badge added in https://github.com/aws-cloudformation/cloudformation-guard/pull/83:
https://github.com/aws-cloudformation/cloudformation-guard/blob/4aa5a0f19c55b0709dcf1d8ef5f02e0d76afee76/README.md#L3

**Pull request merged**